src/internal/util/UnsubscriptionError.ts

```diff
@@ -21,5 +23,17 @@ UnsubscriptionErrorImpl.prototype = Object.create(Error.prototype);
 /**
  * An error thrown when one or more errors have occurred during the
  * `unsubscribe` of a {@link Subscription}.
+ *
+ * NOT INTENDED TO BE CREATED BY CONSUMING CODE.
+ *
+ * @class UnsubscriptionError
+ *
+ * @deprecated (gone in v8) for `instanceof` checks, instead use {@link isTeardownError}
  */
-export const UnsubscriptionError: UnsubscriptionErrorCtor = UnsubscriptionErrorImpl as any;
\ No newline at end of file
+export const UnsubscriptionError: UnsubscriptionErrorCtor = UnsubscriptionErrorImpl as any;
+
+export function createTeardownError(errors: any[]) {
+  const error: any = new UnsubscriptionError(errors);
+  error.__rxjsErrorCode = RxErrorCode.Teardown;
```

Comment:
Is there a reason for not using `createRxError` here?

Reply:
Yes. Temporarily, I needed to leverage the existing one, because of the behavior it has in the constructor. Adding a TODO.